source/core_http_client.c

```diff
@@ -1203,12 +1204,17 @@ static HTTPStatus_t parseHttpResponse( HTTPParsingContext_t * pParsingContext,
 
     /* This will begin the parsing. Each of the callbacks set in
      * parserSettings will be invoked as parts of the HTTP response are
-     * reached. The return code is parsed in #processLlhttpError so is not needed. */
+     * reached. The return code is parsed in #processLlhttpError. */
+    eReturn = llhttp_execute( &( pParsingContext->llhttpParser ), parsingStartLoc, parseLen );
 
-    if( llhttp_execute( &( pParsingContext->llhttpParser ), parsingStartLoc, parseLen ) == HPE_PAUSED_UPGRADE )
+    if( eReturn == HPE_PAUSED_UPGRADE )
     {
         llhttp_resume_after_upgrade( &( pParsingContext->llhttpParser ) );
     }
+    else
+    {
+        /* Empty else MISRA 15.7 */
```

Comment:
MISRA 15.7 does not apply here as there is no `else if` chain., so this should be removed.  

As an aside, even if it were applicable, simply mentioning that the else was added to comply with the rule is against of the rule; such required comments should explain why no code is needed in that case.